### PR TITLE
Connect api

### PR DESCRIPTION
**Pull request by AlaaAlsalem**
Here are the things I added to this repo
> Connect to API
Modify add and remove action

### CODE REVIEW

**Review by RolandM99 — COMMENTED** [review]
Hi @AlaaAlsalem, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ❌

You have submitted a project, that has failed the Eslint checker. Linter result should be green, this means that the last commit has a green check mark(✔️). Has requested by the previous reviewer, Kindly fix this issue :
image


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/components/AllBooks.js [hunk 1]
```diff
@@ -1,39 +1,31 @@
...
+  useEffect(() => {
+    dispatch(getBooks());
+  }, []);
 
   if (myBooks.length) {
     return (
       <div>
-        <Book books={myBooks} propsToRemoveBooks={deleteOneBook} />
-        <BookInput propsToAddBooks={addNew} />
+        <Book books={myBooks} />
```
Comment (akucintavalent) [inline src/components/AllBooks.js:19]: I'd advice you to rename this component to `Books` in order to be consistent

File: src/components/Book.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  return (
+    <ul>
+      {books.map((book) => (
+        <li key={uuidv4()}>
+          <span>{book.title}</span>
+          <br />
+          <span>{book.category}</span>
+          <br />
+          <button type="button" onClick={() => dispatch(deleteBook(book.id))}>Remove</button>
+        </li>
```
Comment (akucintavalent) [inline src/components/Book.js]: You forgot the author of the book.

File: src/components/BookInput.js [hunk 3]
```diff
@@ -1,28 +1,32 @@
...
-    author: '',
-  });
+const categories = ['Choose a Category', 'Action', 'Action and Adventure', 'Drama', 'Mystery', 'Children Books', 'Romance', 'Comedy', 'Finances', 'Accounting', 'Novel', 'Horror', 'Science', 'Health', 'History', 'Business'];
+
+const BookInput = () => {
+  const dispatch = useDispatch();
+  const [title, setTitle] = useState('');
+  const [category, setCategory] = useState(categories[0]);
 
   const onChange = (e) => {
```
Comment (akucintavalent) [inline src/components/BookInput.js]: Kindly rename this method for its purpose to be obvious. You can call it `onBookTitleChange`, for example.

File: src/components/BookInput.js [hunk 4]
```diff
@@ -1,28 +1,32 @@
...
-  const [input, setInput] = useState({
-    title: '',
-    author: '',
-  });
+const categories = ['Choose a Category', 'Action', 'Action and Adventure', 'Drama', 'Mystery', 'Children Books', 'Romance', 'Comedy', 'Finances', 'Accounting', 'Novel', 'Horror', 'Science', 'Health', 'History', 'Business'];
+
+const BookInput = () => {
+  const dispatch = useDispatch();
+  const [title, setTitle] = useState('');
+  const [category, setCategory] = useState(categories[0]);
```
Comment (akucintavalent) [inline src/components/BookInput.js]: You forgot the author.

File: src/components/BookInput.js [hunk 5]
```diff
@@ -1,28 +1,32 @@
...
-      [e.target.name]: e.target.value,
-    });
+    setTitle(e.target.value);
+  };
+
+  const book = {
+    item_id: uuidv4(),
+    title,
+    category,
   };
```
Comment (akucintavalent) [inline src/components/BookInput.js]: You forgot about the author.

File: src/components/BookInput.js [hunk 6]
```diff
@@ -1,28 +1,32 @@
...
-      propsToAddBooks(input.title, input.author);
-      setInput({
-        title: '',
-        author: '',
-      });
+    if (title && category) {
+      dispatch(postBooks(book));
+      setTitle('');
+      setCategory(categories[0]);
+      e.target.reset();
```
Comment (akucintavalent) [inline src/components/BookInput.js]: You forgot about the author.

File: src/components/BookInput.js [hunk 7]
```diff
@@ -34,27 +38,20 @@ const BookInput = (props) => {
         <input
```
Comment (akucintavalent) [inline src/components/BookInput.js:45]: Kindly add the input field for the author.

**Review by akucintavalent — CHANGES_REQUESTED** [review]
REQUIRED CHANGES

Hi there @AlaaAlsalem, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### To highlight :tada: :champagne: 
- Nice PR description and title :+1: 
- Nice README file :heavy_check_mark:
- GitHub flow is followed :+1:


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by RolandM99 — APPROVED** [review]
<h1>Status: Approve 🥇 ✔</h1>
Hi @AlaaAlsalem, 

### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**